packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-checkbox-group/gux-form-field-checkbox-group.service.ts

```diff
@@ -0,0 +1,52 @@
+import simulateNativeEvent from '@utils/dom/simulate-native-event';
+
+export function setAllCheckboxInputs(
+  root: HTMLElement,
+  checked: boolean
+): void {
+  getCheckboxInputs(root).forEach(checkboxInput => {
+    if (checkboxInput.checked !== checked && !checkboxInput.disabled) {
+      checkboxInput.checked = checked;
+      simulateNativeEvent(checkboxInput, 'input');
+      simulateNativeEvent(checkboxInput, 'change');
+    }
+  });
+}
+
+export function setMainCheckboxElementCheckedState(
```

Comment:
Minor but would this be better named as `setParentCheckboxElementCheckedState` to match the design.